### PR TITLE
Initial pass on updating documentation to include MonoVM

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

File: docs/workflow/README.md [hunk 1]
```diff
@@ -46,33 +46,37 @@ You may need to build the tree in a combination of configurations. This section
...
-* **CoreLib** (also known as System.Private.CoreLib - code under src/coreclr/System.Private.CoreLib) -- this is the lowest level managed library. It has a special relationship with the runtime -- it must be in the matching configuration, e.g., if the runtime you are using was built in a debug configuration, this must be in a debug configuration.
-* **All other libraries** (most code under src/libraries) -- the bulk of the libraries are oblivious to the configuration that CoreCLR/CoreLib were built in. Like most code they are most debuggable when built in a debug configuration, and, happily, they still run sufficiently fast in that configuration that it's acceptable for development work.
+* **Runtime** is the execution engine for managed code and there are two different runtimes available. Both runtimes are written in C/C++, therefore, easier to debug when built in a Debug configuration.
+    * CoreCLR VM is the primary desktop runtime which if built in Debug Configuration it executes managed code very slowly. For example it will take a long time to run the managed code unit tests. The code lives under [src/coreclr](../../src/coreclr).
```
Comment (jkotas) [inline docs/workflow/README.md]: Nit: The previous paragraph talks about different runtimes, and this one switches to VM. We should be consistent.

Comment (jkotas) [inline docs/workflow/README.md]: When we say "desktop runtime", we mean .NET Framework. Calling CoreCLR a desktop runtime will just cause confusion. Also, CoreCLR is portable - it is just less portable than Mono.

We may want a separate paragraph or document that describes where each runtime runs and what each runtime is  is best for.

Comment (marek-safar) [reply docs/workflow/README.md]: Changed the wording a bit

File: docs/workflow/testing/mono/testing.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Running Tests on MonoVM
```
Comment (jkotas) [inline docs/workflow/testing/mono/testing.md]: Nit: This is the only place that says MonoVM. Should it say just Mono?

**Review by jkotas — APPROVED** [review]

**Pull request merged**